### DISCUSSION TITLE
Ubuntu 20.04LTS Metacity client screen lock - can't be unlocked

### DISCUSSION THREAD

**Discussion by expertnoob** [ltsp/ltsp#499] [General]
Hello everyone,
Has anyone encountered an issue with Metacity (Gnome Flashback) where after a user's screen goes into locked mode after idling or from a manual lock-screen action, it cannot be unlocked even after entering the correct password? (The same password used to login successfully in the first place.)

The LTSP client image is a "chroot" image that is trimmed down Ubuntu 20.04 LTS Desktop with  gdm3, ubuntu-desktop, gnome-shell and metacity installed.

On the client side, I have confirmed that the user login works on the tty2 console as well.

When entering the correct password at the unlock screen, the interface show the error:

> Sorry, that didn't work. Please try again

, and in the LTSP server side's auth.log:

```
Jul 20 11:20:15 ltsp-testsvr sshd[64719]: Accepted password for tester from 192.168.1.190 port 41122 ssh2
Jul 20 11:20:15 ltsp-testsvr sshd[64719]: pam_unix(sshd:session): session opened for user tester by (uid=0)
Jul 20 11:20:15 ltsp-testsvr systemd-logind[471]: New session 273 of user tester.

Jul 20 11:20:16 ltsp-testsvr sshd[64784]: Received disconnect from 192.168.1.190 port 41122:11: disconnected by user
Jul 20 11:20:16 ltsp-testsvr sshd[64784]: Disconnected from user tester 192.168.1.190 port 41122
Jul 20 11:20:16 ltsp-testsvr sshd[64719]: pam_unix(sshd:session): session closed for user tester
Jul 20 11:20:16 ltsp-testsvr systemd-logind[471]: Session 273 logged out. Waiting for processes to exit.
Jul 20 11:20:16 ltsp-testsvr systemd-logind[471]: Removed session 273.
```

If the wrong password is entered on purpose, the client side unlock-screen would show ".Login incorrect." and then "Sorry, that didn't work.Please try again"

I have also checked in the /home/<user>/.xsession-error file, and it is an empty file.

Any suggestion for troubleshooting this issue?

NOTE: Using Ubuntu's "Ubuntu" or "Ubuntu Wayland" desktop manager, the unlock screen works correctly, and the server side auth.log shows similar log entries:

```
Jul 20 13:43:09 ltsp-testsvr sshd[112247]: pam_unix(sshd:session): session opened for user tester by (uid=0)
Jul 20 13:43:09 ltsp-testsvr systemd-logind[471]: New session 328 of user tester.
Jul 20 13:43:09 ltsp-testsvr sshd[112299]: Received disconnect from 192.168.1.190 port 55952:11: disconnected by user
Jul 20 13:43:09 ltsp-testsvr sshd[112299]: Disconnected from user tester 192.168.1.190 port 55952
Jul 20 13:43:09 ltsp-testsvr sshd[112247]: pam_unix(sshd:session): session closed for user tester
Jul 20 13:43:09 ltsp-testsvr systemd-logind[471]: Session 328 logged out. Waiting for processes to exit.
Jul 20 13:43:09 ltsp-testsvr systemd-logind[471]: Removed session 328.
```

----
Update/Extra Info:

* xfce4's lock screen works

**Comment by alkisg** [comment]
Try to write that chroot in a local disk, with grub etc. Boot it without using LTSP, log in with a local user and see if lock screen works.
If not, maybe you're missing some dependencies; you'd need to check with the metacity authors for searching which ones.

**Comment by alkisg** [comment]
Actually just add a local user (with useradd) while the client is running, and test with that one, as it won't use pamltsp.

**Reply by expertnoob** [reply]
I have tested with adding a local user (named it "localuser") in the client computer; however, the user can't be used to login with Metacity selected.  However, the localuser can login with the default 'ubuntu' or 'ubuntu wayland' desktop manager.

When Metacity is selected, the screen goes blank for a while after login, and then returns back to the main users listing login page.

**Comment by alkisg** [comment]
It sounds like some things are missing from your metacity installation.
A "minimal chroot" isn't really lighter than other chroots, as the whole virtual disk isn't loaded in RAM, the necessary sectors are fetched on demand.
I would suggest that you use a normal installation, e.g. chrootless, VM, or even chroot cloned from a VM, to make sure that all the necessary packages are there.

**Reply by expertnoob** [reply]
I was looking to restrict some applications for client using chroot images..

Ok. I will be testing out the normal installation, chrootless image.
